float64 x
float64 y
float64 z
float64 xt
float64 yt
float64 zt
bool validation
---
bool success